### PR TITLE
Vulnerability fixes: Change base image and allow newer versions of fastapi and uvicorn

### PR DESCRIPTION
**Pull request by chiragjn**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+RUN microdnf install -y \
+  unzip \
+  tar \
+  xz \
+  unzip \
+  git \
+  jq \
+  ca-certificates \
+  python3.11-devel python3.11-pip \
+  && ln -s /usr/bin/python3.11 /usr/bin/python3 \
```
Comment (debajyoti-truefoundry) [inline Dockerfile:13]: Hmm, this is interesting.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+  && ln -s /usr/bin/python3 /usr/bin/python \
+  && ln -s /usr/bin/pip3.11 /usr/bin/pip
```
Comment (debajyoti-truefoundry) [inline Dockerfile:15]: `/usr/bin/pip3` too? 🙈

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,18 @@
-FROM python:3.9-slim
-ENV DEBIAN_FRONTEND=noninteractive
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.4-1227.1726694542
```
[the remark this answers is not in the document]
Comment (chiragjn) [reply Dockerfile:1]: For vulnerabilities in debian libcrypto and ssl packages

File: Dockerfile (hunk 2 above)
Comment (chiragjn) [reply Dockerfile:15]: Sadly on redhat yes

**Pull request merged**